.github/workflows/ci_cd.yml

```diff
@@ -0,0 +1,175 @@
+name: CI + CD
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+      - name: Cache pre-commit envs
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pre-commit
+          key: ${{ runner.os }}-pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Run pre-commit to lint files
+        run: |
+          pip install pre-commit
+          pre-commit run --all-files
```

Comment:
Just occurred to me that we can use this action to run pre-commit 😱 

https://github.com/pre-commit/action

Reply:
That seems to be deprecated as per the `README.md`, which suggests using https://pre-commit.ci and that site doesn't look very reliable.

Reply:
Ahhh I had looked into pre-commit.ci before in order to get the PR comments feature. I think it is fine, but not worth switching to just for this.